sdk/formrecognizer/ai-form-recognizer/src/models.ts

```diff
@@ -256,78 +256,201 @@ export interface FieldData {
-      value?: string;
-      valueType?: "phoneNumber";
-    }
-  | {
-      value?: number;
-      valueType?: "integer";
-    }
-  | {
-      value?: FormField[];
-      valueType?: "array";
-    }
-  | {
-      value?: Record<string, FormField>;
-      valueType?: "object";
-    }
-  | {
-      value?: SelectionMarkState;
-      valueType?: "selectionMark";
-    }
-  | {
-      value?: string;
-      valueType?: "gender";
-    }
-  | {
-      value?: string;
-      valueType?: "country";
-    }
-);
+}
+
+/**
+ * A catch-all form field variation with an unknown value type.
+ *
+ * This interface is provided for type safety and should only be encountered
+ * when the `valueType` of a FormField is undefined, and there is no ordinary
+ * reason that should be the case.
+ */
+export interface FormUnknownField extends FormFieldCommon {
+  valueType?: undefined;
+  /**
+   * If `valueType` is undefined, then the type of the value is unknown.
+   */
+  value?: unknown;
+}
+
+/**
+ * A form field with a string value.
+ */
+export interface FormStringField extends FormFieldCommon {
+  valueType: "string";
+  /**
+   * The value of the recognized string.
+   */
+  value?: string;
+}
+
+/**
+ * A form field with a numeric value.
+ */
+export interface FormNumberField extends FormFieldCommon {
+  valueType: "number";
+  /**
+   * The value of the recognized number.
+   */
+  value?: number;
+}
+
+/**
+ * A form field with a value representing a date.
+ */
+export interface FormDateField extends FormFieldCommon {
+  valueType: "date";
+  /**
+   * The value of the date field, represented as a JavaScript date object.
+   */
+  value?: Date;
+}
+
+/**
+ * A form field with a value representing a time.
+ */
+export interface FormTimeField extends FormFieldCommon {
+  valueType: "time";
+  /**
+   * The value of the time field, represented as a string.
+   */
+  value?: string;
+}
+
+/**
+ * A form field with a value representing a phone number.
+ */
+export interface FormPhoneNumberField extends FormFieldCommon {
+  valueType: "phoneNumber";
+  /**
+   * The value of the recognized phone number, represented as a string.
+   *
+   * This value is normalized to a uniform string representation, e.g.
+   * "+447911123456". If the service is unable to normalize the value,
+   * information about the phone number's textual appearance may appear in the
+   * `text` property of the value data (see the `valueData` field of this
+   * object).
+   */
+  value?: string;
+}
+
+/**
+ * A form field with an integer value.
+ */
+export interface FormIntegerField extends FormFieldCommon {
+  valueType: "integer";
+  /**
+   * The value of the recognized integer.
+   */
+  value?: number;
+}
+
+/**
+ * A form field with an array of FormFields as a value.
+ */
+export interface FormArrayField extends FormFieldCommon {
+  valueType: "array";
+  /**
+   * The recognized array of nested fields. Each value in this array is its
+   * own `FormField`.
+   */
+  value?: FormField[];
```

Comment:
Will the FormField's in the array be generic or have specific FormTypeFields returned?

Reply:
`FormField` is defined to be the union of all of the variants, so it could be any one of them. This basically maps 1:1 with the syntactic schema of the service's form field responses.